data/virt_autotest/guest_console_monitor.sh

```diff
@@ -0,0 +1,284 @@
+			        done
+
+                    sleep_time=180
+                    sleep_count=0
+                    failed_count=0
+                    unset vmguest_check_failed
+                    declare -a vmguest_check_failed=""
+                    while [ ${sleep_count} -le ${sleep_time} ];do
+                        echo -e "Try $((${sleep_count}/10 + 1)) time(s) check\n"
+                        vmguest_index=0
+
+                        #Check each vm guest and store checked result.
+                        for vmguest in ${vm_guestnames_array[@]};do
+                            #ping -c 1 ${vmguest}
+                            virsh list --state-running | grep ${vmguest}
+                            vmguest_check_failed[${vmguest_index}]=`echo $?`
+                            vmguest_index=$((${vmguest_index} + 1))
+                        done
+                        failed_count=0
+                        vmguest_index=0
+                    
+                        #Iterate vmguest_check_failed to check whether ping to each vm guest succeeded.
+                        for vmguest in ${vm_guestnames_array[@]};do
+                            if [[ ${vmguest_check_failed[${vmguest_index}]} -ne 0 ]];then
+                                echo -e "Connection to ${vmguest} failed !\n"
+                                failed_count=$((${failed_count} + 1))
+                            else
+                                echo -e "Connection to ${vmguest} succeeded !\n"
+                            fi
+                            vmguest_index=$((${vmguest_index} + 1))
+                        done
+                    
+                        #If ping to all vm guests succeeded, terminate while loop and proceed. 
+                        if [[ ${failed_count} -eq 0 ]];then
+                            echo -e "Connection to all vm guests successfully.\n"
+                            break
+                        fi
+                        sleep_count=$((${sleep_count} + 10))
+                        sleep 10s
+                    done			        
+			        
+			        #Check with background PID of virsh console
+			        for vmguest in ${vm_guestnames_array[@]};do
+			            virsh list --state-running | grep ${vmguest}
+			                if [[ $? -eq 0 ]];then
+			                    ps aux | grep "virsh console ${vmguest}"
+                                if [[ $? -ne 0 ]];then
+                                    get_console ${vmguest} >> ${LOG_DIR}/${vmguest}_console_${timestamp} 2>&1 &
+                                fi
+			                else			                    
+			                    virsh list --inactive | grep ${vmguest}
+			                    if [[ $? -eq 0 ]];then
+			                        echo "${vmguest} was shutdown"
+			                        echo "No any output from virsh console ${vmguest}"
+			                    fi                            
+			             fi
+			        done
```

Comment:
I think this check logic is enough. No need that "wait and sleep loop".

Reply:
removed "wait and sleep loop".